src/components/action-popover/action-popover.test.js

```diff
@@ -0,0 +1,702 @@
+          <ActionPopoverCustom />
+        </Accordion>
+      );
+
+      accordionDefaultTitle().trigger("keydown", keyCode("Enter"));
+      actionPopoverButton().eq(0).click();
+      actionPopover()
+        .should("have.attr", "data-popper-placement", "bottom-end")
+        .and("be.visible");
+      cy.scrollTo("0", "0");
+      actionPopover()
+        .should("have.attr", "data-popper-placement", "top-end")
+        .and("be.visible");
+    });
+  });
+
+  describe("check props for ActionPopover component", () => {
+    it("should render ActionPopover with unique id", () => {
+      CypressMountWithProviders(<ActionPopoverWithProps id="cypress" />);
+
+      actionPopoverButton().eq(0).click();
+      actionPopoverWrapper().should("have.attr", "id", "cypress");
+    });
+
+    it.each([
+      [true, "right"],
+      [false, "left"],
+    ])(
+      "should render ActionPopover with rightAlignMenu set to %s",
+      (rightAlignMenu, css) => {
+        CypressMountWithProviders(
+          <ActionPopoverWithProps rightAlignMenu={rightAlignMenu} />
+        );
+
+        actionPopoverButton().eq(0).click();
+        actionPopover().should("have.css", css);
+      }
+    );
+
+    it("should render ActionPopover with custom button", () => {
+      CypressMountWithProviders(
+        <ActionPopoverWithProps
+          renderButton={() => (
+            <ActionPopoverMenuButton
+              buttonType="tertiary"
+              iconType="dropdown"
+              iconPosition="after"
+              size="small"
+            >
+              More
+            </ActionPopoverMenuButton>
+          )}
+        />
+      );
+
+      buttonDataComponent().click();
+      actionPopoverWrapper()
+        .find('[data-component="button"]')
+        .should("be.visible");
+    });
+
+    it.each([
+      ["left", "start"],
+      ["right", "end"],
+    ])(
+      "should render ActionPopover with horizontalAlignment prop set to %s",
+      (position, attrValue) => {
+        CypressMountWithProviders(
+          <ActionPopoverWithProps horizontalAlignment={position} />
+        );
+
+        actionPopoverButton().eq(0).click();
+        actionPopover()
+          .children()
+          .should("have.css", "justify-content", `flex-${attrValue}`);
+      }
+    );
+
+    it("should render ActionPopoverMenu with menuID", () => {
+      CypressMountWithProviders(
+        <ActionPopoverMenuWithProps menuID="cypress" />
+      );
+
+      actionPopoverButton().eq(0).click();
+      actionPopover().eq(1).should("have.attr", "id", "cypress");
+    });
+
+    it("should render ActionPopoverMenu with focusIndex", () => {
+      CypressMountWithProviders(
+        <ActionPopoverMenuWithProps focusIndex={null} />
+      );
+
+      actionPopoverButton().eq(0).click();
+      actionPopover().eq(1).children().eq(0).should("not.be.focused");
+    });
+  });
+
+  describe("Check events for ActionPopover component", () => {
```

Comment:
Should we also test the onOpen and onClose (popover opened and popover closed) events?

Reply:
added